content/blog/eng/lambda-days-2020.md

```diff
@@ -0,0 +1,91 @@
+type like non empty arrays that otherwise a compiler would not know
+
+[Video](https://www.youtube.com/watch?v=3zT5eVHpQwA&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=13&t=0s)
+
+- **Designing composable functional libraries, not just for data visualization**
+
+[Video](https://www.youtube.com/watch?v=G1Dp0NtQHeY&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=11&t=0s)
+
+**Tomas Petricek** from the Alan Turing Institute showed his tools library for data journalism
+He presented this library written in `f#`, and with a live demo built some graphs highlighting the endless possibilities given from the composability.
+
+
+- **Enhancing the type system with Refined Types**
+
+[Video](https://www.youtube.com/watch?v=Fx8WXcAZWuk&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=28&t=0s)
+
+**Juliano Alves**, Senior Software Engineer @TransferWise talked about refined types in Scala
+In this talks aimed to Scala beginners, he talked about how is achieve better return types.
+Through [Refined](https://github.com/fthomas/refined) library, which is a porting from the homonymus Haskell one, he showed how is possible
+constrain types using predicates and create types like below zero number and so on.
+
+
+- **Effect Handlers: A New Approach to Computational Effects**
+
+[Video](https://www.youtube.com/watch?v=6lv_E-CjGzg&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=37&t=0s)
+
+**Maciej Piróg** is a researcher working on design and semantics of programming languge
+His talk hinge on managing effects with `Effect handlers` instead of Monads, comparing the pros and cons of both approaches.
+Then he showed with a live coding session, how `Helium` language can handle easily and in a functional manner effects without loosing typings.
+This sounds interesting albeit `Helium` is far from being a solid production ready programming language.
+
+- **One-shot algebraic effects as coroutines**
+
+[Video](https://www.youtube.com/watch?v=JQwc1OBOt5k&list=PLvL2NEhYV4ZsV9Bw0wp1P46SOdtk4pFW6&index=22&t=0s)
+
+**Satoru Kawahara** is a student at University of Tsukuba
+Exceptions can't resume computations, while algebraic effects can. The latter, and their advantages, are not available on commercial programming languages, therefore 
+some of the advantages they provide are not available in everyday programming.
+Satoru and his collagues had an intuition about using one-time algebraic effects implemented as coroutines, thorugh resumable exceptions you can catch it then contnue
+computation (only one time) as if it you were using a free monad (on which they were implemented).
+This intuition is particulary powerful because is easy to implement (there are already many libraries based on this) in most programming languages.
+
+### Conclusions
+
+The Lambda Days is maybe one of the most important conference in the functional world and we were surprised by how well organized was; this is only a brief
```

Comment:
Lambda Days is probably one of the most important conferences in the functional world and we were impressed by the excellent organization. In this article, we only provided a brief list of what we found particularly interesting. Therefore, we suggest you to visit the conference website and see for yourself.